enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,511 @@
+
+**Limitations**:
+- No automatic IP conflict detection across multiple management clusters
+- No integration with external IPAM systems
+
+**Improvements**:
+- If we manage to merge the [PR](https://github.com/coreos/ignition/pull/2134) to use NoCloud with ignition instead of ConfigDrive
+  then the network data format can be cloud init network config v1 or v2 and also JSON or yaml, this is more human friendly than
+  current opnstack config format.
+
+### Risks and Mitigations
+
+**Risk**: IP address conflicts if multiple clusters use overlapping IP pools
+**Mitigation**: Document best practices for IP pool planning. Consider future enhancement for IP conflict detection or integration with external IPAM solutions.
+
+**Risk**: Incorrect network configuration could make nodes unreachable
+**Mitigation**: Provide comprehensive validation of network configuration in the API. Document testing procedures and troubleshooting steps. Ensure console access is available for debugging boot issues.
+
+**Risk**: Afterburn changes could affect other CoreOS deployments
+**Mitigation**: Changes are isolated to KubeVirt provider detection and handling. Extensive testing in CI for both HyperShift and standalone CoreOS deployments.
+
+### Drawbacks
+
+- Adds complexity to the hypershift kubevirt controllers:
+  - implementing and ip pool mechanism
+  - Need to parse and understand the openshift network config format
+- Fix depend on a coreos change:
+  - Change should be accepted and merged by non hypershift related team.
+  - It will take some time for the RPM to reach RHCOS, also we will have to plan backports there.
+
+## Alternatives (Not Implemented)
+
+### Store all the ignition config directly at a secret instead of retrieve it using the ignition server
+
+Currently hypershift workers retrieve the ignition config from an ignition server, so they need to have network configured before that, an alternative
+would be to directly put the whole ignition config directly at the secret referenced by the VM so accessing ignition server is not needed.
+
+**Why not chosen**: The kubernetes resources has a size limit that has already being reache by hypershift when storing the whole ignition config there [Jira](https://issues.redhat.com//browse/OCPBUGS-60148), also it means changing how a hypershift fundamental part of the bootstrap process works.
+
+### Use the ovn-kubernetes localnet IPAM mechanism and implement the missing gateway and dns part
+
+With this solution IPAM configuration will be done at the ovn-kubernetes localnet user defined network instead of the hypershift kubevirt nodepool, but ovn-kubernetes is missing features for it:
+- There is no official support for localnet + IPAM
+- localnet is missing gateway and dns configuration
+
+Also this makes customer depend on ovn-kubernetes and localnet, while the solution on this enhancement makes it CNI independant, working even with other CNIs like cillium or calico.
```

Comment:
I don't think this would just work with cillium and calico. How about saying it will work with SR-IOV or linux bridge CNI